hadoop-hdds/rocks-native/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedSSTDumpIterator.java

```diff
@@ -86,51 +79,47 @@ public ManagedSSTDumpIterator(ManagedSSTDumpTool sstDumpTool,
    * @return Optional of the integer empty if no integer exists
    */
   private Optional<Integer> getNextNumberInStream() throws IOException {
-    StringBuilder value = new StringBuilder();
-    int val;
-    while ((val = processOutput.read()) != -1) {
-      if (val >= '0' && val <= '9') {
-        value.append((char) val);
-      } else if (value.length() > 0) {
-        break;
-      }
+    int n = processOutput.read(intBuffer, 0, 4);
+    if (n == 4) {
```

Comment:
Is it considered as data corruption if bytes read are in between 1-3?

Reply:
yeah

Reply:
Ideally it should not be corrupted, since rocksdb checks the sst file before reading it.

Reply:
If that's the case, then we should throw an`IllegalStateException` if it happens and fail the request instead of returning `Optional.empty();`. I think we can keep returning `Optional.empty();` when there is no more byte (n is either 0 or -1)  to read.

Reply:
Done. Have added a test case for this as well.

Reply:
Thanks